pkg/utils/netmonitor/netmonitor.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2019-2024 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package netmonitor
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+// Options defines the options for the monitoring.
+type Options struct {
+	Link  bool
+	Addr  bool
+	Route bool
+}
+
+// StartMonitoring starts the monitoring of the network interfaces.
+// If there is a change in the network interfaces, it will send a message to the channel.
+// With the options, you can choose to monitor only the link, address, or route changes.
```

Comment:
Can we have a default behaviour if the options are nil? (and describe it in the comment)

It can be:

* return an error
* set all to true
* set all to false
* other

Reply:
I setted all to true as default